src/controllers/controllerruntimedata.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <QVariant>
+
+#include "util/assert.h"
+
+/// ControllerRuntimeData is a wrapper that allows controllers script runtimes
+/// to share arbitrary data via a the JavaScript interface. It doesn't enforce
+/// any type consistency and it is the script responsibility to use this data in
+/// a considerate, non-destructive way (append to lists, extend to objects,
+/// ...), as well as expecting that others won't do so.
+class ControllerRuntimeData : public QObject {
```

Comment:
For me the term `ControllerRuntimeData` is not self-descriptive. I would suggest to call it `ControllerSharedData` instead.